### PR TITLE
Remove pcre dependency

### PR DESCRIPTION
**Pull request by sdbondi**
I've removed the pcre dependency (therefore, the need for CGO and the libpcre++ dependency). All tests passing

### CODE REVIEW

**Comment by jiejie-dev** [comment]
I'm waiting for this PR.

Thank for what you have done ! @sdbondi

File: unmarshal_test.go [hunk 1]
```diff
@@ -6,7 +6,12 @@ import (
 )
 
 func TestUnmarshal(t *testing.T) {
-	hash, err := Unmarshal("foo")
+	hash, err := Unmarshal("")
+	if assert.NoError(t, err) {
+		assert.Equal(t, hash, map[string]interface{}{})
+	}
```
Comment (derekstavis) [inline unmarshal_test.go:12]: Good addition!! 👍

**Comment by derekstavis** [comment]
That's awesome!! Thanks!!

Merging as it seems that it's all good!! ✨

**Comment by sdbondi** [comment]
@derekstavis Oh one thing I didn't do, I didn't remove pcre from the `glide.lock` file as I'm not familiar with glide. Can you remove from the lock file and make a tag? :smile:

**Comment by derekstavis** [comment]
Hey, sorry for the delay, I've removed from glide and published tag `0.3.0`!! Thanks!!

**Pull request merged**